app/controllers/bookmarks_controller.rb

```diff
@@ -31,6 +31,10 @@ def update
     @user = current_user
     @topic = @user.topics.find(params[:topic_id])
     @bookmark = @topic.bookmarks.find(params[:id])
+    
+    # is this even necessary?
```

Comment:
@ianroberts131 Good point, glad you brought it up. 

The scoping queries that ensure a user is authorized to edit only their own bookmarks do provide a very similar protection to what Pundit is doing, so it could be argued as unnecessary. 

However, with data security, more layers of defense are desirable, even if there is some overlap/repetion. See https://en.wikipedia.org/wiki/Layered_security and https://en.wikipedia.org/wiki/Defense_in_depth_(computing).